kanidm_book/src/installing_the_server.md

```diff
@@ -5,41 +5,93 @@ Currently we have a pre-release docker image based on git master. They can be fo
     https://hub.docker.com/r/kanidm/server
     https://hub.docker.com/r/kanidm/radius
 
-You'll need a volume where you can put certificates and the database:
+You can fetch these with:
+
+    docker pull kanidm/server:latest
+    docker pull kanidm/radius:latest
+
+There are a number of configuration steps you must perform before you can run this container.
+
+## TLS
+
+You'll need a volume where you can place configuration, certificates and the database:
 
     docker volume create kanidmd
 
-You should have a ca.pem, cert.pem and key.pem in your kanidmd volume. The reason for requiring
-TLS is explained in [why tls](./why_tls.md) . To put the certificates in place you can use a shell container
-that mounts the volume such as:
+You should have a chain.pem and key.pem in your kanidmd volume. The reason for requiring
+TLS is explained in [why tls](./why_tls.md). In summary TLS is our root of trust between the
+server and clients, and a critical element of ensuring a secure system.
+
+The key.pem should be a single PEM private key, with no encryption. The file content should be
+similar to:
+
+    -----BEGIN RSA PRIVATE KEY-----
+    MII...
+    ...
+    -----END RSA PRIVATE KEY-----
+
+The chain.pem is a series of PEM formatted certificates. The leaf certificate, or the certificate
+that matches the private key should be the first certificate in the file. This should be followed
+by the series of intermediates, and the final certificate should be the CA root. For example:
+
+    -----BEGIN CERTIFICATE-----
+    <leaf certificate>
+    -----END CERTIFICATE-----
+    -----BEGIN CERTIFICATE-----
+    <intermediate certificate>
+    -----END CERTIFICATE-----
+    [ more intermediates if needed ]
+    -----BEGIN CERTIFICATE-----
+    <ca/croot certificate>
+    -----END CERTIFICATE-----
+
+You can validate that the leaf certificate is matching to the key with the command:
+
+    # openssl rsa -noout -modulus -in key.pem | openssl sha1
+    d2188932f520e45f2e76153fbbaf13f81ea6c1ef
+    # openssl x509 -noout -modulus -in chain.pem | openssl sha1
+    d2188932f520e45f2e76153fbbaf13f81ea6c1ef
+
+You can validate the chain to the certificate with:
+
+    openssl verify -CAfile chain.pem chain.pem
```

Comment:
This seems not to work on a Let's Encrypt certificate.

Steps I use to generate chained certificate for kanidm.
```
# cd /etc/letsencrypt/live/<FQDN>
# cat cert.pem chain.pem > cert_chain.pem
# openssl verify -CAfile cert_chain.pem cert_chain.pem
C = US, O = Let's Encrypt, CN = R3
error 2 at 1 depth lookup: unable to get issuer certificate
error cert_chain.pem: verification failed
```

Reply:
Good catch! I investigated why, and I have updated the documentation to suit - I have also tested on my owns lets encrypt certs and validated the steps are correct now :)